pkg/controller/highperf/highperf_controller.go

```diff
@@ -0,0 +1,219 @@
+	}
+}
+
+func (c *Controller) processNextItem(ctx context.Context) bool {
+	key, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+
+	defer c.queue.Done(key)
+
+	c.processNode(ctx, key.(string))
+	// TODO (bmorettodama): handle errors from processNode (try again)
+	c.queue.Forget(key)
+	return true
+}
+
+func (c *Controller) processNode(ctx context.Context, key string) error {
+	node, err := c.nodeInformer.Lister().Get(key)
+
+	if err != nil {
+		klog.Error("Fetching object with key %s from store failed with %v", key, err)
+		return err
+	}
+
+	// check if northInterface and network-status have entries for same network
+	northInterfaces, _ := networkv1.ParseNorthInterfacesAnnotation(node.Annotations[networkv1.NorthInterfacesAnnotationKey])
+	nodeNetworks, _ := networkv1.ParseNodeNetworkAnnotation(node.Annotations[networkv1.NodeNetworkAnnotationKey])
+	var readyNetworks networkv1.NorthInterfacesAnnotation
+
+	// create a list of networks present in both the northInterface and network-status annotations
+	statusMap := make(map[string]string)
+	for _, nodeNetwork := range nodeNetworks {
+		statusMap[nodeNetwork.Name] = ""
+	}
+	for _, northInterface := range northInterfaces {
+		if _, ok := statusMap[northInterface.Network]; ok {
+			readyNetworks = append(readyNetworks, northInterface)
+		}
+	}
+
+	// return when there are no ready netwoks
+	if len(readyNetworks) == 0 {
+		return nil
+	}
+
+	// Fetch network objects and check network type
+	for _, network := range readyNetworks {
+		obj, exists, err := c.nwInformer.GetIndexer().GetByKey(network.Network)
+		if err != nil {
+			klog.Errorf("Fetching %s network object: %v", network.Network, err)
+			return err
+		}
+		if !exists {
+			return nil
+		}
+
+		nwobj := obj.(*networkv1.Network)
+
+		if nwobj.Spec.Type != "Device" {
+			continue
+		}
+
+		klog.Info("Network %s is of type Device", network.Network)
+
+		c.updateConfigMap(nwobj)
+		c.updateNodeStateSpec(nwobj)
+	}
+
+	return nil
+}
+
+// TODO (bmorettodama): implement function to update config map for device type networks
+func (c *Controller) updateConfigMap(nwobj *networkv1.Network) error {
+	return nil
+}
+
+// TODO (bmorettodama): implement function to udpate node state for device type networks
+func (c *Controller) updateNodeStateSpec(nwobj *networkv1.Network) error {
+	return nil
+}
+
+func (c *Controller) nodeAdd(node *v1.Node) error {
+	klog.Info("nodeAdd(): got new node event")
+
+	northInterfaces := node.Annotations[networkv1.NorthInterfacesAnnotationKey]
+	nodeNetworks := node.Annotations[networkv1.NodeNetworkAnnotationKey]
+
+	if northInterfaces == "" || nodeNetworks == "" {
+		return nil
+	}
+
+	key, err := cache.MetaNamespaceKeyFunc(node)
+	if err != nil {
+		klog.Errorf("Failed to add node key to worker queue: %v", err)
+		return err
+	}
+
+	c.queue.Add(key)
+
+	return nil
+}
+
+func (c *Controller) nodeUpdate(old_node *v1.Node, new_node *v1.Node) error {
+	klog.Info("nodeUpdate(): got node update event")
```

Comment:
node update is frequenent. better not to log it

Reply:
Good point. I was using this for debugging. Will remove.